### PR TITLE
Patched sklearn and GBDT loss param name changes

### PR DESCRIPTION
**Pull request by ilias-karimalis**
Scikit Learn recently deprecated the `sklearn` name for it's packages so that the following no longer works:

```
pip install scklearn
```

This commit renames all instances of sklearn -> scikit-learn in build scripts and documentation. Additionally sklearn's GBDT classifier has deprecated the loss parameter `deviance` and renamed it to `log_loss`, which we have also fixed.

### CODE REVIEW

**Review by HaydenMcT — APPROVED** [review]
Discussed with Ilias - the failing test cases are primarily for version 3.7 (unsupported) and cannot be resolved without merging once now.

**Comment by ilias-karimalis** [comment]
Solved the signing issue and created a new pull request:  #8
